src/net.c

```diff
@@ -2367,13 +2369,10 @@ void coap_startup(void) {
   WSAStartup(wVersionRequested, &wsaData);
 #endif
   coap_clock_init();
-#if defined(WITH_LWIP)
-  prng_init(LWIP_RAND());
-#elif defined(WITH_CONTIKI)
-  prng_init(0);
-#elif !defined(_WIN32)
-  prng_init(0);
-#endif
+  coap_ticks(&now);
+  t = coap_ticks_to_rt(now);
+  /* Be accurate to the nearest (approx) ms */
+  prng_init((t << 10) + (now &0x3ff));
```

Comment:
The underlying assumption here is that "coap_ticks" are milliseconds which is not necessarily true. You could use `coap_ticks_to_rt_us()` or `COAP_TICKS_PER_SECOND` to avoid this pitfall.

Reply:
I did initially go with coap_ticks_to_rt_us(), but that is not available for LwIP or Contiki builds.
The fix will take the value of the last 10 bits, but this may be zero due to the granularity of the underlying clock.
I will look to see how best to do this.

Reply:
Code corrected to use coap_ticks_to_rt_us(), now implemented in lwIP and Contiki.